app/src/main/java/com/andryoga/safebox/data/db/secureDao/UserDetailsDaoSecure.kt

```diff
@@ -3,14 +3,17 @@ package com.andryoga.safebox.data.db.secureDao
 import com.andryoga.safebox.data.db.dao.UserDetailsDao
 import com.andryoga.safebox.data.db.entity.UserDetailsEntity
 import com.andryoga.safebox.security.interfaces.HashingUtils
+import com.andryoga.safebox.security.interfaces.SymmetricKeyUtils
 import javax.inject.Inject
 
 class UserDetailsDaoSecure @Inject constructor(
     private val userDetailsDao: UserDetailsDao,
-    private val hashingUtils: HashingUtils
+    private val hashingUtils: HashingUtils,
+    private val symmetricKeyUtils: SymmetricKeyUtils
 ) : UserDetailsDao {
     override suspend fun insertUserDetailsData(userDetailsEntity: UserDetailsEntity) {
         userDetailsDao.insertUserDetailsData(hash(userDetailsEntity))
+        userDetailsDao.insertUserDetailsData(encrypt(userDetailsEntity))
```

Comment:
you are calling **userDetailsDao.insertUserDetailsData(.....)** two times, this will either add two rows in db or it will fail or it will cause data issue in db